domains/pallets/messenger/src/lib.rs

```diff
@@ -287,12 +293,11 @@ mod pallet {
     pub(super) type OutboxResponses<T: Config> =
         StorageValue<_, Message<BalanceOf<T>>, OptionQuery>;
 
-    /// A temporary storage to store all the messages to be relayed in this block.
-    /// Will be cleared on the initialization on next block.
+    /// Storage to store the weight tags for all the outbox and inbox response messages.
     #[pallet::storage]
-    #[pallet::getter(fn block_messages)]
-    pub(super) type BlockMessages<T: Config> =
-        StorageValue<_, crate::messages::BlockMessages, OptionQuery>;
+    #[pallet::getter(fn message_weight_tags)]
+    pub(super) type MessageWeightTags<T: Config> =
+        StorageValue<_, crate::messages::MessageWeightTags, OptionQuery>;
```

Comment:
IMHO it is weird to see this separate storage for storing the message weight tag, how about storing the weight tag along with the message instead? either add a field to the `Message` struct or store `(Message, MessageWeightTag)` in the `Outbox` and `InboxResponses` storage.

Reply:
I thought about this. Adding this to the Message would mean increasing the size the storage proof while this tag is already part of the wraped XDM. I would like to keep it seperate for now

Reply:
>Adding this to the Message would mean increasing the size the storage proof while this tag is already part of the wraped XDM

Since the Message already contains this tag we can remove the `weight_tag` field from the wrapped XDM.

Reply:
Not necessarily. Spent yesterday doing this.
Turns out we need to read the read the storage to get the message weight tag which does not seems right. 
Also turns out domain bundle weight checks are stateless while this change requires them to be stateful where we need to run a pre-dispatch.
IMO, weights should be at max be derivable from the extrinsic arguments instead of reading from the storage. The change you are proposing would need to read from the storage post xdm validation since the weight tag will not be present in the xdm proof itself. Will we are storing during the pre dispatch.

For now, lets leave it as is and note that weights should never require any form of stateful storage

Reply:
I'm not following, the expression in `#[pallet::weight(...)]` is stateless, the weight tag should be part of the extrinsic (constructed by the src chain) not read from the state of the dst chain.

If it is tricky to handle I'm okay to leave it as is but plz open an issue to track it so we can revisit it later.

Reply:
It ideally would not be once the weight tag is moved to Inner message which is inside the storage proof.
So in order to get the weight tag, we need to verify XDM and get the weight tag. This is currently being done in pre_dispatch where we verify, extract and set to Inbox or OutboxResponse. This would be a storage call post pre dispatch. 
Other option would be to extract proof without using storage which I dont believe is possible. Its not tricky but I dont think this would even work.